### ISSUE TITLE
Improve tests

### ISSUE THREAD

**Issue by mariobehling** [fossasia/searss#26]
As mentioned elsewhere by @rhemon 

> it would better if you could improve the tests. such as testing if results collected are matching a pattern or not? also it might be more organised if you had each of the classes in different files. and have all of the tests in a tests directory. you will see lot of repositories follow this pattern. try a larger coverage increase.
... it should only be concentrating on rss-generator.py but there are other files part of it as well. we might need to look into that.

**Comment by In-vincible** [comment]
@rhemon should I start writing tests, I have throughly tested the working I am ready to add tests :) Also I would like to discuss, more thorough testing, as in right now we are only testing if we are getting response with non-zero length, is it enough?

**Comment by rhemon** [comment]
@In-vincible yes please go ahead and add tests. u can ask for help anytime. once you add the tests u thought of, do a Pull Request. if we think something else can be added or changed we can mention it there.

**Comment by In-vincible** [comment]
Cool! so I should add tests in tests.py only?

**Comment by rhemon** [comment]
@In-vincible you can do that, or else if you want you can have a test folder containing a TestSuite and different test file for the different search engines used, also for the other helper function used.

**Comment by In-vincible** [comment]
I have added tests for duckduckgo and askcom, I am into adding tests for generateFeed function should I give a PR at this stage for discussion purpose?

**Comment by In-vincible** [comment]
Also I would love to do lots of other scraping, I am aware with selenium as well, do we have any more plans for searss @mariobehling  @rhemon

**Comment by In-vincible** [comment]
Should we add unittest-xml module for XML testing?

**Comment by rhemon** [comment]
yes you can do a PR so that we can discuss.
and it would be good if you use unittest xml, this will be needed to reach
100% coverage

and i would say you can go ahead and add whatever you want to searss and do
a PR for that. then we can discuss and improve that as well. or if you want
first create a issue for the feature u want to add. we discuss there first
and then you can add?

**Comment by In-vincible** [comment]
Okay I am putting up the PR, while locally I have am setting up the unittest-xml for testing the xml patterns, we will discuss the xml things on the PR then :)
